extensions/ql-vscode/src/vscode-tests/no-workspace/remote-queries/variant-analysis-monitor.test.ts

```diff
@@ -18,35 +18,55 @@ import { processFailureReason } from '../../../remote-queries/variant-analysis-p
 import { Credentials } from '../../../authentication';
 import { createMockVariantAnalysis } from '../../factories/remote-queries/shared/variant-analysis';
 import { VariantAnalysisManager } from '../../../remote-queries/variant-analysis-manager';
+import { createMockExtensionContext } from '..';
+import { CodeQLCliServer } from '../../../cli';
+import { logger } from '../../../logging';
+import { createMockCliServer } from './factories/cli';
 
 describe('Variant Analysis Monitor', async function() {
   this.timeout(60000);
 
   let sandbox: sinon.SinonSandbox;
-  let extension: CodeQLExtensionInterface | Record<string, never>;
+  let cli: CodeQLCliServer;
   let mockGetVariantAnalysis: sinon.SinonStub;
   let cancellationTokenSource: CancellationTokenSource;
   let variantAnalysisMonitor: VariantAnalysisMonitor;
   let variantAnalysis: VariantAnalysis;
   let variantAnalysisManager: VariantAnalysisManager;
   let mockGetDownloadResult: sinon.SinonStub;
+  let commandSpy: sinon.SinonStub;
 
   beforeEach(async () => {
     sandbox = sinon.createSandbox();
     sandbox.stub(config, 'isVariantAnalysisLiveResultsEnabled').returns(false);
 
+    commandSpy = sandbox.stub(commands, 'executeCommand');
+
     cancellationTokenSource = new CancellationTokenSource();
 
     variantAnalysis = createMockVariantAnalysis();
 
     try {
-      extension = await extensions.getExtension<CodeQLExtensionInterface | Record<string, never>>('GitHub.vscode-codeql')!.activate();
-      variantAnalysisMonitor = new VariantAnalysisMonitor(extension.ctx);
+      const ctx = createMockExtensionContext();
+      cli = createMockCliServer({
+        bqrsInfo: [{ 'result-sets': [{ name: 'result-set-1' }, { name: 'result-set-2' }] }],
+        bqrsDecode: [{
+          columns: [{ kind: 'NotString' }, { kind: 'String' }],
+          tuples: [['a', 'b'], ['c', 'd']],
+          next: 1
+        }, {
+          columns: [{ kind: 'String' }, { kind: 'NotString' }, { kind: 'StillNotString' }],
+          tuples: [['a', 'b', 'c']]
+        }]
+      }, sandbox);
+      const storagePath = path.join(__dirname);
+
+      variantAnalysisMonitor = new VariantAnalysisMonitor(ctx);
+      variantAnalysisManager = new VariantAnalysisManager(ctx, cli, storagePath, logger);
```

Comment:
I'm not sure what exactly we're testing on this instance. The `VariantAnalysisMonitor` doesn't have a reference to the `VariantAnalysisManager` at all, nor to this specific `variantAnalysisManager`. We only have expectations on `expect(mockGetDownloadResult).to.not.have.been.called` and don't have the equivalent for `expect(mockGetDownloadResult).to.have.been.called` so wouldn't notice if it doesn't call this at all. Would it make sense to completely remove this and the expectations?

Reply:
Yeah, in retrospect keeping the setup for the CLI and variant analysis manager was wishful thinking (e.g. could we still use these in the future for integration tests?). But at the moment they're not used properly so I've removed them: https://github.com/github/vscode-codeql/pull/1628/commits/ad19d0d06de2217c7df7bef33a77acd2fc117a7a